src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -529,6 +529,11 @@ def delete(
                 )
             )
 
+    def _update_hosts(hosts_info_resp):
+        if hosts_info_resp.get('allow_any_host'):
+            hosts_info_resp['hosts'].insert(0, {"nqn": "*"})
+        return hosts_info_resp
```

Comment:
Although the difference is very subtle, I preferred the former approach, as it kept the final representation hidden by operating with models (classes and properties, like `Host(nqn="*")` and `Host.allow_any_host`) instead of dicts (`host_info_resp.get('allow_any_host')` and `{"nqn": "*"}`). Not a big deal anyway.

Reply:
yes I think so as well, but I am afraid we dont have much choice here since now the pick decorator is used mostly for the api and not the cli. 
I just hope it is a step backward for two steps forward :)